public/app/plugins/panel/canvas/InlineEditOptions.tsx

```diff
@@ -0,0 +1,99 @@
+import { get as lodashGet } from 'lodash';
+import React, { useMemo } from 'react';
+import { useObservable } from 'react-use';
+
+import { PanelOptionsEditorBuilder, StandardEditorContext } from '@grafana/data';
+import { PanelOptionsSupplier } from '@grafana/data/src/panel/PanelPlugin';
+import { NestedValueAccess } from '@grafana/data/src/utils/OptionsUIBuilders';
+import { GroupState } from 'app/features/canvas/runtime/group';
+import { OptionsPaneCategoryDescriptor } from 'app/features/dashboard/components/PanelEditor/OptionsPaneCategoryDescriptor';
+import { fillOptionsPaneItems } from 'app/features/dashboard/components/PanelEditor/getVisualizationOptions';
+import { setOptionImmutably } from 'app/features/dashboard/components/PanelEditor/utils';
+
+import { activePanelSubject, InstanceState } from './CanvasPanel';
+import { getElementEditor } from './editor/elementEditor';
+import { getLayerEditor } from './editor/layerEditor';
+
+export const InlineEditOptions = () => {
+  const activePanel = useObservable(activePanelSubject);
+  const instanceState = activePanel?.panel.context?.instanceState;
+
+  const pane = useMemo(() => {
+    const state: InstanceState = instanceState;
+    if (!state) {
+      return new OptionsPaneCategoryDescriptor({ id: 'root', title: 'root' });
+    }
+
+    const supplier = (builder: PanelOptionsEditorBuilder<any>, context: StandardEditorContext<any>) => {
+      builder.addNestedOptions(getLayerEditor(instanceState));
+
+      const selection = state.selected;
+      if (selection?.length === 1) {
+        const element = selection[0];
+        if (!(element instanceof GroupState)) {
+          builder.addNestedOptions(
+            getElementEditor({
+              category: [`Selected element (${element.options.name})`],
+              element,
+              scene: state.scene,
+            })
+          );
+        }
+      }
+    };
+
+    return getOptionsPaneCategoryDescriptor({}, supplier);
+  }, [instanceState]);
+
+  return (
+    <div>
+      <div>{pane.items.map((v) => v.render())}</div>
+      <div>
+        {pane.categories.map((c) => {
+          return (
+            <div key={c.props.id}>
+              <h5>{c.props.title}</h5>
```

Comment:
For title would it potentially make more sense for this to be the panel's title (i.e. `Panel Title Editor`, maybe even `Panel Title Canvas Editor` to be more explicit I guess) instead of hardcoded to `Canvas Inline Editor`?

Reply:
Lets punt on that for now.  Hardcoded container title is fine for our needs for a while